platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -330,9 +326,7 @@ private File getTemporaryDirWithoutCreating() {
      * @since 6.4
      */
     @Nested
-    public ModularitySpec getModularity() {
-        return modularity;
-    }
+    public abstract ModularitySpec getModularity();
```

Comment:
This won't be quite the same because DefaultModularSpec also sets a convention on inferModulePath